lib/facility_factory.rb

```diff
@@ -0,0 +1,46 @@
+class FacilityFactory
+  def initialize
+  end
+
+  def create_facility(api)
+    api.map do |facility|
+      Facility.new({
+        name: format_name(facility),
+        address: format_address(facility),
+        phone: format_phone_number(facility) 
+      })
+    end
+  end
+
+  def format_name(facility)
+    if facility[:state] && facility[:state] == "MO"
```

Comment:
Hmm, I don't think you need to check both of these things actually. If the :state key didn't exist, then `facility[:state] == "MO"` would just return false since `nil == "MO"` would be false. So you just need `if facility[:state] == "MO"`